Configurations/50-nonstop.conf

```diff
@@ -0,0 +1,176 @@
+#### Nonstop configurations
+    "nonstop-common" => {
+        inherit_from     => [ "BASE_unix" ],
+        template         => 1,
+        cc               => "c99",
+        cflags           => add_before(picker(debug   => "-g -O0",
+                                              release => "-g -O2") ,"-Wextensions -Wnowarn=203,220,272,734,770,1506 -Wbuild_neutral_library"),
+        perl             => "/usr/bin/perl",
+        lflags           => "-lrld",
+        shared_target    => "self",
+        shared_cflag     => "",
+        shared_ldflag    => "-Wshared",
+        shared_extension => ".so",
+        enable           => ["egd"],
+        dso_scheme       => "DLFCN",
+    },
+    "nonstop-nsx" => {
+        inherit_from     => [ "nonstop-common" ],
+        cflags           => sub { join(" ",@_,"-Wverbose -I/usr/local/include") },
+        shared_cflag     => sub { join(" ",@_,"-Wverbose -I/usr/local/include") },
+        lflags           => sub { join(" ",@_,"-lfloss -Wxld='-set floattype neutral_float' -Wsystype=oss") },
+        shared_ldflag    => sub { join(" ",@_,"-lfloss -Wxld='-export_all -set floattype neutral_float -set systype oss' -Wsystype=oss") },
```

Comment:
Just like `shared_cflag` is a complement for `cflags`, `shared_ldflag` is a complement for `lflags`, i.e. it shouldn't be necessary to repeat...  however, seeing the additions in the latter, I gotta ask if `-Wxld` is a cumulative flag, or if it's rather a "last man standing" type of flag?  The latter could prove problematic.

Furthermore, if `-export_all` does what I think it does, I gotta ask, is there no way to select which symbols are going to be exported?

Reply:
Will fix up the `shared_cflag` and `cflag` usage as well as `shared_ldflag` and `ldflag`.

The `-Wxld` is similar to the `-Wrld`. Loader option. I don't think it is positional. Will check.

The `-export_all` does what you think it does, but only exports globals. static variables and procedures are not exported. There is a keyword in the platform's c99 (`export$`) that could be used instead but it is on a per-symbol basis in C source. I have not successfully used that. Otherwise it is possible to selectively specify each symbol on the command line, or in a linker input file - they we get into the same problem as `ar` and `.so` construction limits.

Reply:
We do have support for linker input files that specify what symbols should be exported...

Reply:
The `-Wxld` is how parameters are passed to the linker through the c99 front end. It is possible to specify multiple copies of this AFAIK.

Reply:
The `-exported_symbol` option to the linker can be specified in a linker input file, it would look something like:

```
-exported_symbol A
-exported_symbol B
etc
```
Passed in through the `-Wxld_obey=file` option.

Reply:
`-exported_symbol` looks useful together with `-Wxld_obey=file`. I can probably concoct something tomorrow that should help.